File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams : Params
    {
        public string CurrentEmail { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; }
        public int MaxAge { get; set; }
        public string OrderBy { get; set; } = "lastActive";
        private string _search;
        public string Search 
        { 
            get => _search;
            set => _search = value.ToLower();
        }
    }
}

File: client/src/app/shared/models/userParams.ts
import { Params } from "./params";
import { User } from "./user";

export class UserParams extends Params {
    gender: string;
    minAge = 18;
    maxAge = 35;
    orderBy = 'lastActive';
    override pageSize = 5;

    constructor(user: User) {
        super();
        this.gender = user.gender === 'female' ? 'male' : 'female'
    }
}
